.github/workflows/nightly.yml

```diff
@@ -0,0 +1,31 @@
+name: Nightly Release
+
+on:
+  schedule:
+    - cron:  '0 9 * * *'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install requirements
+      run: |
+        pip install -r dev-requirements.txt
+    - name: Build Package
+      env: 
+        ALLENNLP_VERSION: $(git rev-parse --short "$GITHUB_SHA")
```

Comment:
I think `git describe` is better for this.

Reply:
I'd argue that the SHA is precisely how we should identify git commits. Let's not introduce another identifier with different semantics.

Reply:
Yeah, but, for example, which version is the latest, `ac19db2` or 5fc3719`?

Reply:
What I mean is that PyPI/pip has to determine an order for the versions, and it gets hard with just the SHA.

Reply:
`git describe` gives you the latest tag from the parents, the number of commits there are from it and the SHA; in that order.

Reply:
Ah, sure. I would expect that a nightly release would be named (in part) off of the date it was made.

Reply:
Here's what I get when I install the torch nightlies, for instance:

```
$ pip list | grep torch
torch       1.5.0.dev20200211
torchvision 0.6.0a0+d8d36f0
```

(The most recent tag for `torchvision` is v0.5.0, incidentally. So this is forward looking in contrast to `git describe`.)

Reply:
```
(tmp) brendanr.local ➜  allennlp git:(master) ✗ git describe
v0.9.0-215-g473db763
(tmp) brendanr.local ➜  allennlp git:(master) ✗ git show g473db763
fatal: ambiguous argument 'g473db763': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

That's not great behavior...

Reply:
I think the leading `g` is extra. Try with the rest.

Reply:
(I think it goes for "git")

Reply:
Good catch, thank you. That works.

Reply:
But I still see no reason to use this. Date is standard for naming nightlies. SHA is standard for identifying git commits. No need for a new thing.

Reply:
```bash
echo $(git describe --abbrev=0).dev$(date +%Y%m%d)
```

Reply:
`date -u +%Y%m%d` to use UTC 0.

Reply:
`echo $(date +%Y%m%d)-$(git rev-parse --short master)`

Reply:
But this version is gonna be greater than the stable releases. Also, for SemVer, you need to specify somewhere that it's "dev" or something like that, otherwise systems will think it's stable.

Reply:
This is just the revision. 0.9.0 gets added inside `version.py`. Thanks for the heads up about "dev".